node/src/components/block_synchronizer/tests.rs

```diff
@@ -225,7 +267,13 @@ impl BlockSynchronizer {
     ) -> BlockSynchronizer {
         let mut block_synchronizer = BlockSynchronizer::new(
             config,
-            Arc::new(Chainspec::random(rng)),
+            Arc::new(Chainspec {
+                core_config: CoreConfig {
+                    legacy_required_finality: LegacyRequiredFinality::Strict,
```

Comment:
There's quite a variation in the historical sync flow with regards to signature sufficiency determination based on this parameter. There's also `CoreConfig::start_protocol_version_with_strict_finality_signatures_required` to take into account which is tightly related to this parameter.
Can you please add some tests to check transitions from HaveWeakFinality and to HaveStrictFinality based on variations of these 2 parameters?

Reply:
I added a test that uses `legacy_required_finality: LegacyRequiredFinality::Weak`.